### PR TITLE
Minor Fixes + NavigationService & custom DrawerNavigator support

### PR DESCRIPTION
**Pull request by levi-beers**
This PR takes into account the latest release of RNS.

1. A minor fix to the header image making it 100% width as it was short on most iOS & Android devices.
Screen Shot 2019-03-26 at 12 45 44 AM

2. RNS utilizing React Navigation v3 has a TabNavigator nested in a StackNavigator, this is fine however by default, it's behavior is to override the TabNavigator titles. If you switch tabs, the title never changes from 'React Native Starter'. I modified TabNavigator, which is now exported as a function, to changes titles on a "switch" case. Users can easily modify their own logic here. 

3. Swapped the "Home" text in HomeView with the RNS logo.

4. Added a custom NavigationService for linking to different screens inside of any app component without passing props. 

Users can

```
import NavigationService from './NavigationService';

        <Drawer.Item
          label="Home"
          icon="home"
          style={styles.screenTextStyle}
          onPress={() => { NavigationService.resetMainNavigator() }}
        />
        <Drawer.Item
          label="Gallery"
          icon="perm-media"
          style={styles.screenTextStyle}
          onPress={() => { NavigationService.goToSubScreen('Dashboard','Gallery') }}
        />
```
NavigationService can be easily modified to include custom functions. It currently contains a nested navigation switch screen function, resetting to home screen, open/close drawer navigator.

Additionally, I added support for DrawerNavigator. In many cases, apps use DrawerNavigator and since RNS is IMHO the best starter template, I gave a shot at including it. The CustomDrawer component utilizes react-native-paper, which RNS already includes. CustomDrawer also references the new NavigationService. 

Screen Shot 2019-03-26 at 1 59 42 AM

### CODE REVIEW

**Comment by levi-beers** [comment]
@gaykov This is rebased off of your latest update to RNS

File: src/modules/navigation/CustomDrawer.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      </View>
+    </SafeAreaView>
+  </ScrollView>
+);
+
+export default CustomDrawerContentComponent;
+
+const styles = StyleSheet.create({
+  container: {
+      flex: 1
```
Comment (gaykov) [inline src/modules/navigation/CustomDrawer.js]: That's weird that eslint didn't detected this, but you have 4 spaces here and below. Can we please go with 2? You can format all the files with prettier and this config: https://github.com/flatlogic/react-native-starter/blob/master/.prettierrc

File: src/modules/navigation/CustomDrawer.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    width: 300,
+    justifyContent: 'center',
+    height: 75,
+    alignItems: 'center',
+  },
+  headerContainer: {
+      height: 150,
+  },
+  headerText: {
+      color: '#fff8f8',
```
Comment (gaykov) [inline src/modules/navigation/CustomDrawer.js]: Please move the color to the `colors` file under styles folder

File: src/modules/navigation/MainTabNavigator.js [hunk 3]
```diff
@@ -161,3 +166,25 @@ export default createBottomTabNavigator(
     },
   },
 );
+
+bottomTabNav.navigationOptions = ({ navigation }) => {
+  const { routeName } = navigation.state.routes[navigation.state.index];
+
+  // Set the headerTitle to the current route name
+  const currentRoute = routeName;
```
Comment (gaykov) [inline src/modules/navigation/MainTabNavigator.js]: I think everything below can be rewritten like:
```
return {
  headerTitle: currentRoute === 'Home' ? 'React Native Starter' : currentRoute,
}
```

File: src/modules/navigation/CustomDrawer.js (hunk 1 above)
Comment (levi-beers) [reply src/modules/navigation/CustomDrawer.js]: That is weird. Should be good now.

File: src/modules/navigation/CustomDrawer.js (hunk 2 above)
Comment (levi-beers) [reply src/modules/navigation/CustomDrawer.js]: Done and done 👍

File: src/modules/navigation/MainTabNavigator.js (hunk 3 above)
Comment (levi-beers) [reply src/modules/navigation/MainTabNavigator.js]: That makes significantly more sense now.

**Comment by gaykov** [comment]
Hi!

I'm sorry this PR didn't land to master as expected. We introduced some of this changes in our latest release, so I'm closing this PR.

Thank you again for contributing!

**Pull request closed without merging**